src/adapter/src/catalog.rs

```diff
@@ -6662,6 +6663,12 @@ impl SessionCatalog for ConnCatalog<'_> {
     fn aws_privatelink_availability_zones(&self) -> Option<BTreeSet<String>> {
         self.state.aws_privatelink_availability_zones.clone()
     }
+
+    fn is_enabled(&self, feature_name: &str) -> bool {
+        let result = self.state.system_configuration.is_true(feature_name);
+        soft_assert!(result.is_ok());
+        result.unwrap_or(false)
```

Comment:
Using `false` as the default seems like the safer option. The `soft_assert` above will hopefully ensure that we catch any dangling `is_enabled` calls to no longer existing feature flags in Sentry.

Reply:
I went with a

```rust
/// Returns `true` iff the given `feature` is supported at the moment.
fn supports_feature(&self, feature: CatalogFeature) -> bool;
```

method where `CatalogFeature` is defined as follows:

```rust
/// Defines features that the [SessionCatalog] may optionally support.
#[derive(Debug)]
pub enum CatalogFeature {
    /// True iff `WITH MUTUALLY RECURSIVE` syntax is enabled.
    EnableWithMutuallyRecursive,
}
```